rfcs/0132-meson-builds-nix.md

```diff
@@ -0,0 +1,135 @@
+
+In light of this state of things, we propose a novel alternative to the current building infrastructure.
+
+We expect to accomplish, among other goals,
+
+- better code structuring and documentation;
+- improved cross-platform support, especially outside NixOS and Linux programming environments
+- shorter build times;
+- improved unit testing;
+- an overall improved user experience.
+
+# Detailed design
+[design]: #detailed-design
+
+A carefully crafted set of Meson files should be included in the Nix repository, providing a description on how to deploy the Nix repository, generating all the expected artifacts (command line tools, libraries, configuration files etc.)
+
+This novel building infrastructure should be able to provide at least feature parity with the current quasi-autotools implementation, albeit in a different user interface.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+Currently @p01arst0rm is writing an implementation from scratch.
+
+Here is a table comparing some expected interactions:
+
+|----------------------|--------------------|------------------------------|
+| Action               | Current            | Meson                        |
+|----------------------|--------------------|------------------------------|
+| Configuring          | `./configure`      | `meson setup build_dir`      |
+| Building             | `make`             | `meson -C build_dir compile` |
+| Installing           | `make install`     | `meson -C build_dir install` |
+| Command-line Options | `--enable-gc=true` | `-Dgc=enabled`               |
+|----------------------|--------------------|------------------------------|
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Some possible drawbacks:
+
+- A new build system would require changes on the code
+  + On the other hand, such changes are likely to improve the code base.
+  
+- A new build system requires the developers become familiarized with it
+  - Specially when this build system uses its own description language.
+  + However, Meson is well documented, and its Python-esque language is easy to grasp.
+
+- A new build system indirectly brings its own dependencies to the Nix project
+  - In particular, the reference implementation of Meson is written in Python.
+  - Further, this reference implementation generates script files meant to be consumed by Ninja.
+  - Ninja is a tool written in C++ that acts like a `make` replacement.
+  - This particular setting brings concerns about complexifying the bootstrap route.
+  + Given that Nix is currently written in C++, we can assume a C++ compiler as part of such a bootstrap route.
+  + There are full-featured alternative tools that replace Meson and Ninja. 
+    + Namely, Muon and Samurai are implementations of Meson and Ninja that require only a C compiler and a set of POSIX standard tools.
+  + Autotools also have its own set of dependencies, and a fair comparison should include them
```

Comment:
```suggestion
  + the current autotools buildsystem also has its own set of dependencies, and a fair comparison should include them
  + such dependencies for autotools are sometimes difficult to obtain on non gnu/linux platforms, and often do not offer feature parity. This can often cause issues when attempting to compile on these platforms.
```

Reply:
I will love to include this, however you should bring evidence. I believe it will not be easy because we have MinGW and MSYS2 after all.

Reply:
ill have to look into it, iirc its another bsd thing